app/controllers/tutoring_sessions_controller.rb

```diff
@@ -29,25 +39,55 @@ def update
       redirect_to @tutoring_session,
                   notice: 'Tutoring session was successfully updated.'
     else
+      set_students_and_tutor
       render :edit
     end
   end
 
   def destroy
     @tutoring_session.destroy
-    redirect_to tutoring_sessions_url,
-                notice: 'Tutoring session was successfully destroyed.'
+    redirect_to tutoring_sessions_path(
+      tutor_id: @tutor.id,
+      notice: 'Tutoring session was successfully destroyed.'
+    )
   end
 
   private
 
   def set_tutoring_session
     @tutoring_session = TutoringSession.find(params[:id])
+    match = Match.find(@tutoring_session.match_id)
+    @tutor = Tutor.of(current_user).find(match.tutor_id)
```

Comment:
Something is quite wrong there. Looks like you don't have permission to see the tutor for which you're creating a tutoring session. What account are you using?